conformance/lang/expressions/unary-expression/unary-minus-expression.balt

```diff
@@ -0,0 +1,893 @@
+Labels: DecimalNumber, int:Signed8, unary-minus
+
+public function main() {
+    int:Signed8 a = -1;
+    int:Signed8 b = -2;
+    int:Signed8 c = -101;
+    int:Signed8 d = -128;
+
+    io:println(-a); // @output 1
+    io:println(-b); // @output 2
+    io:println(-c); // @output 101
+    io:println(-d); // @output 128
+}
+
+Test-Case: output
+Description: Test unary negative expression for positive int:Signed16, no overflow scenarios.
+Labels: DecimalNumber, int:Signed16, unary-minus
+
+public function main() {
+    int:Signed16 a = 1;
+    int:Signed16 b = 2;
+    int:Signed16 c = 12345;
+    int:Signed16 d = 32767;
+
+    io:println(-a); // @output -1
+    io:println(-b); // @output -2
+    io:println(-c); // @output -12345
+    io:println(-d); // @output -32767
+}
+
+Test-Case: output
+Description: Test unary negative expression for negative int:Signed16, no overflow scenarios.
+Labels: DecimalNumber, int:Signed16, unary-minus
+
+public function main() {
+    int:Signed16 a = -1;
+    int:Signed16 b = -2;
+    int:Signed16 c = -12345;
+    int:Signed16 d = -32768;
+
+    io:println(-a); // @output 1
+    io:println(-b); // @output 2
+    io:println(-c); // @output 12345
+    io:println(-d); // @output 32768
+}
+
+Test-Case: output
+Description: Test unary negative expression for positive int:Signed32, no overflow scenarios.
+Labels: DecimalNumber, int:Signed32, unary-minus
+
+public function main() {
+    int:Signed32 a = 1;
+    int:Signed32 b = 2;
+    int:Signed32 c = 40123;
+    int:Signed32 d = 2147483647;
+
+    io:println(-a); // @output -1
+    io:println(-b); // @output -2
+    io:println(-c); // @output -40123
+    io:println(-d); // @output -2147483647
+}
+
+Test-Case: output
+Description: Test unary negative expression for negative int:Signed32, no overflow scenarios.
+Labels: DecimalNumber, int:Signed32, unary-minus
+
+public function main() {
+    int:Signed32 a = -1;
+    int:Signed32 b = -2;
+    int:Signed32 c = -40123;
+    int:Signed32 d = -2147483648;
+
+    io:println(-a); // @output 1
+    io:println(-b); // @output 2
+    io:println(-c); // @output 40123
+    io:println(-d); // @output 2147483648
+}
+
+Test-Case: output
+Description: Test unary negative expression for positive user defined sub types of int,
+              no overflow scenarios.
+Labels: DecimalNumber, module-type-defn, unary-minus, union-type
+
+type Ints -2|-1|0|1|2;
+
+public function main() {
+    Ints a = 1;
+    Ints b = 2;
+    Ints c = 0;
+
+    io:println(-a); // @output -1
+    io:println(-b); // @output -2
+    io:println(-c); // @output 0
+}
+
+Test-Case: output
+Description: Test unary negative expression for negative user defined sub types of int,
+              no overflow scenarios.
+Labels: DecimalNumber, module-type-defn, unary-minus, union-type
+
+type Ints -2|-1|0|1|2;
+
+public function main() {
+    Ints a = -1;
+    Ints b = -2;
+
+    io:println(-a); // @output 1
+    io:println(-b); // @output 2
+}
+
```

Comment:
Shall we also add some tests like 
```ballerina
type Ints int:Signed8|int:Unsigned32;

public function main() {
    Ints a = 12;
    int:Signed8|int:Unsigned32 b = 12;

    io:println(-a); // operator not defined
    io:println(-b); // operator not defined
}
```
They are also not working atm.

Reply:
I will add this, Have created similar issue for this also. Unary negation does not work with user defined sub types

Reply:
updated in 6862c07